applications/PfemFluidDynamicsApplication/custom_processes/adaptive_time_interval_process.hpp

```diff
@@ -284,163 +284,148 @@ namespace Kratos
 	OpenMPUtils::PartitionedIterators(mrModelPart.Elements(),ElemBegin,ElemEnd);
 	for ( ModelPart::ElementIterator itElem = ElemBegin; itElem != ElemEnd; ++itElem )
```

Comment:
this loop is not parallelized! you should use _#pragma omp for_ right before the 'for'

Reply:
I found several places in the PfemFluid where the code is in a parallel section, but the workload of the loop is not distributed among the threads.
_#pragma omp parallel_ starts a new parallel region, but if you miss 'for', the loop is not distributing its load, and all threads do the same work (all elements are computed by all threads!).

Reply:
Ok, thank you. Please, see the new commit and check if it looks right!

Reply:
I was totally wrong! The line
OpenMPUtils::PartitionedIterators(mrModelPart.Nodes(),NodeBegin,NodeEnd);
is assigning the correct values to NodeBegin and NodeEnd, so you are parallelizing it manually. That is totally fine. I am sorry about the waste of time.